src/utils/contracts/sputnik-dao.ts

```diff
@@ -0,0 +1,167 @@
+import { tx, view } from "../wallet";
+
+
+const FACTORY_ADDRESS_SELECTOR: Record<string, string> = {
+    "mainnet": "sputnik-dao.near",
+    "testnet": "sputnikv2.testnet"
+};
+// base URLs for SputnikDAO reference UI. See: https://github.com/near-daos/sputnik-dao-2-ui-reference
+const REFERENCE_UI_URL_SELECTOR: Record<string, string> = {
+    "mainnet": "https://v2.sputnik.fund/#",
+    "testnet": "https://testnet-v2.sputnik.fund/#"
+};
+// base URLs for AstroDAO UI. See: https://github.com/near-daos/astro-ui
+const ASTRO_UI_URL_SELECTOR: Record<string, string> = {
+    "mainnet": "https://app.astrodao.com",
+    "testnet": "https://testnet.app.astrodao.com"
+};
+// what SputnikDAO UIs are supported? (P.S.: do NOT use const or string enum here)
+enum SputnikUI {
+    REFERENCE_UI,
+    ASTRO_UI
+}
+// SputnikDAO contract mapping: version <-> code hashes
+const CONTRACT_CODE_HASHES_SELECTOR: Record<string, string[]> = {
+    "mainnet": [
+        "8RMeZ5cXDap6TENxaJKtigRYf3n139iHmTRe8ZUNey6N", // v2.0
+        "8jmhwSkNeAWCRqsr2KoD7d8BzDYorEz3a3iHuM9Gykrg", // v2.1 (with gas fix)
+        "2Zof1Tyy4pMeJM48mDSi5ww2QQhTz97b9S8h6W6r4HnK" // v3.0
+    ],
+    "testnet": [
+        "8RMeZ5cXDap6TENxaJKtigRYf3n139iHmTRe8ZUNey6N", // 2.0
+        "8LN56HLNjvwtiNb6pRVNSTMtPgJYqGjAgkVSHRiK5Wfv", // v2.1 (with gas fix)
+        "783vth3Fg8MBBGGFmRqrytQCWBpYzUcmHoCq4Mo8QqF5" // v3.0
+    ]
+}
+
+
+class SputnikDAO {
+
+    static FACTORY_ADDRESS: string = FACTORY_ADDRESS_SELECTOR[window.NEAR_ENV];
+    static REFERENCE_UI_BASE_URL: string = REFERENCE_UI_URL_SELECTOR[window.NEAR_ENV];
+    static ASTRO_UI_BASE_URL: string = ASTRO_UI_URL_SELECTOR[window.NEAR_ENV];
+    static CONTRACT_CODE_HASHES: string[] = CONTRACT_CODE_HASHES_SELECTOR[window.NEAR_ENV];
+    
+    DAO_ADDRESS: string;
+
+    constructor(dao_address: string) {
+        this.DAO_ADDRESS = dao_address;
+    }
+
+    /**
+     * check of given accountId is a sputnikDAO instance.
+     * uses code_hash of the contract deployed on accountId.
+     * 
+     * @param accountId 
+     */
+    static async isSputnikDAO (accountId: string): Promise<boolean> {
```

Comment:
This is nice! Please use this to determine the `noDao` error in dao.js

Reply:
will be handled in #118